devicetypes/smartthings/rgbw-light.src/rgbw-light.groovy

```diff
@@ -232,53 +265,54 @@ def setHue(value) {
 
 def setColor(value) {
 	log.debug "setColor($value)"
-	def result = []
+	def rgb
+	if (state.staged == null) {
+		state.staged = [:]
+	}
 	if (value.hex) {
-		def c = value.hex.findAll(/[0-9a-fA-F]{2}/).collect { Integer.parseInt(it, 16) }
-		result << zwave.switchColorV3.switchColorSet(red:c[0], green:c[1], blue:c[2], warmWhite:0, coldWhite:0)
+		state.staged << [color: value.hex] // stage ST RGB color attribute
+		def hsv = colorUtil.hexToHsv(value.hex) // convert to HSV
+		state.staged << [hue: hsv[0], saturation: hsv[1]] // stage ST hue and saturation attributes
+		rgb = value.hex.findAll(/[0-9a-fA-F]{2}/).collect { Integer.parseInt(it, 16) } // separate RGB elements for zwave setter
 	} else {
-		def rgb = huesatToRGB(value.hue, value.saturation)
-		result << zwave.switchColorV3.switchColorSet(red: rgb[0], green: rgb[1], blue: rgb[2], warmWhite:0, coldWhite:0)
+		state.staged << value.subMap("hue", "saturation") // stage ST hue and saturation attributes
+		def hex = colorUtil.hsvToHex(Math.round(value.hue) as int, Math.round(value.saturation) as int) // convert to hex
+		state.staged << [color: hex] // statge ST RGB color attribute
+		rgb = colorUtil.hexToRgb(hex) // separate RGB elements for zwave setter
 	}
-	if (device.currentValue("switch") != "on") {
-		result << zwave.basicV1.basicSet(value: 0xFF)
-		result << zwave.switchMultilevelV3.switchMultilevelGet()
-	}
-	result += queryAllColors()
+	commands([zwave.switchColorV3.switchColorSet(red: rgb[0], green: rgb[1], blue: rgb[2], warmWhite: 0, coldWhite: 0),
+	          zwave.switchColorV3.switchColorGet(colorComponent: RGB_NAMES[0]), // event-publish callback is on any of the RGB responses, so only need to GET one of these
+	], 3500)
```

Comment:
I think this too could be made into a helper function (like with the `on()` and `off()` calls) since the code is somewhat re-used below.

Reply:
I decided to leave this since the opportunity for reuse was less than the multiLevelSet.  Reusing / wrapping the multiLevelSet was a lot more straight forward.